src/App.tsx

```diff
@@ -1,17 +1,132 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { addTodo, deleteTodo, getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList/TodoList';
+import { TempTodo } from './components/TempTodo/TempTodo';
+import { Footer } from './components/Footer/Footer';
+import { ErrorNotify } from './components/ErrorNotification/ErrorNotification';
+
+import { ErrorMessage } from './types/ErrorMessage';
+import { ErrorData } from './types/ErrorData';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState<ErrorData>({
+    status: false,
+    notification: ErrorMessage.None,
+  });
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
```

Comment:
`todos` and `visibleTodos` - do not store in state data which could be calculated by another data
use `useMemo` for visibleTodos

Reply:
corrected